relay-event-schema/src/protocol/ourlog.rs

```diff
@@ -154,90 +159,319 @@ impl Empty for AttributeValue {
 impl FromValue for AttributeValue {
     fn from_value(value: Annotated<Value>) -> Annotated<Self> {
         match value {
-            Annotated(Some(Value::String(value)), meta) => {
-                Annotated(Some(AttributeValue::StringValue(value)), meta)
-            }
-            Annotated(Some(Value::I64(value)), meta) => {
-                Annotated(Some(AttributeValue::IntValue(value)), meta)
+            Annotated(Some(Value::Object(mut object)), meta) => {
+                let attribute_type = object
+                    .remove("type")
+                    .and_then(|v| v.value().cloned())
+                    .and_then(|v| match v {
+                        Value::String(s) => Some(s),
+                        _ => None,
+                    });
+
+                let value = object.remove("value");
+
+                match (attribute_type.as_deref(), value) {
```

Comment:
My rust power level is too low to know whether there is a better way than cloning, open to suggestions